terraform/variable.tf

```diff
@@ -0,0 +1,138 @@
+# AWS related variables - Modify region and instance type as per your requirement
+variable "region" {
+  description = "Newsletter automation ec2 instance deploy region"
+  type        = string
+  default     = "us-east-1"
+}
+
+variable "newsletter_ec2_inst_type" {
+  description = "declaring ec2 instance type for newsletter for test"
+  default     = "t3.small"
+}
+
+variable "keyname" {
+  type        = string
+  default     = "newsletter-key-pair"
+  description = "Key-pair generated by Terraform"
+}
+# Modify default value as per your requirement
+variable "profile" {
+  default = "personal"
+  type    = string
+
+}
+# path to temp key folder name
+variable "temp_path" {
+  type        = string
+  description = "Path to the PEM private key file"
+  default = "/tmp"
```

Comment:
Can we create a new directory, instead of specifying '/tmp'. What if the temp directory is not present for some reason. Instead can we `mkdir` a directory first and specify that?

Reply:
1. The terraform configuration file runs on UNIX flavor so by default all  UNIX or LINUX  will have '/tmp' folder. So we don't need to change that.